templates/phaser/packages/client/src/ui/hooks/useNetworkLayer.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useMemo } from "react";
+import { createNetworkLayer } from "../../layers/network/createNetworkLayer";
+import { usePromiseValue } from "./usePromiseValue";
+
+export const useNetworkLayer = () => {
+  const networkLayerPromise = useMemo(() => {
+    return createNetworkLayer();
```

Comment:
re: above, might be able to just call `setup()` here instead of `createNetworkLayer()`
(note that this may change the shape of data stored here though)